src/pysdccc/_cli.py

```diff
@@ -0,0 +1,133 @@
+import contextlib
+import io
+import pathlib
+import subprocess
+import sys
+import tempfile
+import zipfile
+
+import httpx
+
+from pysdccc import _download, _runner
+
+try:
+    import click
+except ImportError as import_error:
+    msg = "Cli not installed. Please install using 'pip install pysdccc[cli]."
+    raise ImportError(msg) from import_error
+
+
+class UrlType(click.ParamType):
+    name = 'url'
+
+    def convert(self, value: str, param: click.Parameter | None, ctx: click.Context | None) -> httpx.URL:
+        try:
+            return httpx.URL(value)
+        except Exception as e:  # noqa: BLE001
+            self.fail(f'{value!r} is not a valid url: {e}', param, ctx)
```

Comment:
add raise?!

Reply:
return would be unreachable

Reply:
As discussed, it is the task of the user to keep this in mind. Check if the implementation of fail is as expected in the used version of click. The risk is low.